tracer-sofa-boot-starter/src/main/java/com/alipay/sofa/tracer/boot/datasource/processor/DataSourceBeanFactoryPostProcessor.java

```diff
@@ -0,0 +1,180 @@
+ */
+package com.alipay.sofa.tracer.boot.datasource.processor;
+
+import com.alipay.common.tracer.core.utils.StringUtils;
+import com.alipay.sofa.tracer.plugins.datasource.SmartDataSource;
+import com.alipay.sofa.tracer.plugins.datasource.utils.DataSourceUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.MutablePropertyValues;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryUtils;
+import org.springframework.beans.factory.ListableBeanFactory;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.*;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+import org.springframework.util.Assert;
+
+import javax.sql.DataSource;
+import java.security.InvalidParameterException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.alipay.common.tracer.core.configuration.SofaTracerConfiguration.TRACER_APPNAME_KEY;
+
+/**
+ * @author qilong.zql
+ * @since 2.2.0
+ */
+public class DataSourceBeanFactoryPostProcessor implements BeanFactoryPostProcessor,
+                                               EnvironmentAware {
+
+    public static final String SOFA_TRACER_DATASOURCE = "s_t_d_s_";
+
+    private Environment        environment;
+
+    @Override
+    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
+                                                                                   throws BeansException {
+        for (String beanName : getBeanNames(beanFactory, DataSource.class)) {
+            if (beanName.startsWith(SOFA_TRACER_DATASOURCE)) {
+                continue;
+            }
+            BeanDefinition dataSource = getBeanDefinition(beanName, beanFactory);
+            if (DataSourceUtils.isDruidDataSource(dataSource.getBeanClassName())) {
+                createDataSourceProxy(beanFactory, beanName, dataSource, getDruidJdbcUrlKey());
+            } else if (DataSourceUtils.isC3p0DataSource(dataSource.getBeanClassName())) {
+                createDataSourceProxy(beanFactory, beanName, dataSource, getC3p0JdbcUrlKey());
+            } else if (DataSourceUtils.isDbcpDataSource(dataSource.getBeanClassName())) {
+                createDataSourceProxy(beanFactory, beanName, dataSource, getDbcpJdbcUrlKey());
+            } else if (DataSourceUtils.isTomcatDataSource(dataSource.getBeanClassName())) {
+                createDataSourceProxy(beanFactory, beanName, dataSource, getTomcatJdbcUrlKey());
+            } else if (DataSourceUtils.isHikariDataSource(dataSource.getBeanClassName())) {
+                createDataSourceProxy(beanFactory, beanName, dataSource, getHikariJdbcUrlKey());
+            }
```

Comment:
It seems `getXXKey()`  can move to `DataSourceUtils`, too. The code will more simple, like 

```
createDataSourceProxy(beanFactory, beanName, dataSource, getJdbcUrlKeyByDataSource());
```

Reply:
it has been moved to DataSourceUtils.